### PR TITLE
[testing] Add most testing via tox.ini

### PR DESCRIPTION
**Pull request by arif-ali**
Consolidate testing, so that things can be run from CLI, CI and various other means in a consistent way by using tox.
    
* Updated all the flake8 tests in the tests folder.
* Added pylint as a neutral test so that we can work on this in stages and have some collaboration on what we test and don't
* The tox tests for unit, stageone, stagetwo testing makes is easier for users to know how to run tests, and not have to do things manually
* Using tox for CI doesn't make sense, as that will create virtual envs and will disregard system/snap based python modules so may not work
    
---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [ ] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?
- [ ] Are all passwords or private data gathered by this PR [obfuscated](https://github.com/sosreport/sos/wiki/How-to-Write-a-Plugin#how-to-prevent-collecting-passwords)?

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+commands =
+    avocado run -p TESTLOCAL=true --test-runner=runner -t foreman tests/product_tests/foreman/
+
+[testenv:nosetests]
+basepython = python3
+deps =
+    {[testenv]deps}
+    nose3
+commands =
+    nosetests -v --with-coverage --cover-package=sos tests/unittests --cover-html
```
Comment (TurboTurtle) [inline tox.ini:54]: We haven't used `nose` in a while, any particular reason behind bringing it back?

**Review by TurboTurtle — COMMENTED** [review]
Mostly tracks, I think. One question follows.

File: tox.ini (hunk 1 above)
Comment (arif-ali) [reply tox.ini:54]: Ah, I picked that one up from the tests that is run while the deb package is being built.

We don't have avocado in our repost yet, and this is some thing @dnegreira is going to correct in this cycle. Once that is done we can move to use avocado natively.

But this does work

**Review by TurboTurtle — APPROVED** [review]

File: tests/cleaner_tests/existing_archive.py [hunk 2]
```diff
@@ -26,34 +27,52 @@ class ExistingArchiveCleanTest(StageTwoReportTest):
...
-        with open(os.path.join(self.tmpdir, '%s-obfuscation.log' % ARCHIVE), 'r') as log:
+        with open(os.path.join(
+                self.tmpdir,
+                f'{ARCHIVE}-obfuscation.log'), 'r') as log:
             for line in log:
                 if "Loaded %s" % ARCHIVE in line:
-                    assert 'as type sos report archive' in line, "Incorrect archive type detected: %s" % line
+                    assert \
+                        'as type sos report archive' in line, \
+                        "Incorrect archive type detected: %s" % line
```
Comment (pmoravec) [inline tests/cleaner_tests/existing_archive.py:42]: Does it matter we dont use f-strings here?

File: tests/collect_tests/help_output_tests.py [hunk 3]
```diff
@@ -88,17 +98,26 @@ def test_cluster_profiles_shown(self):
         # this has the secondary effect of enforcing a stylistic requirement
         # where at least one profile must match the pyfile name
         for clus in clusters:
-            assert clus in _profs, "Cluster '%s' not displayed in --list-options" % clus
+            assert \
+                clus in _profs, \
+                "Cluster '{clus}' not displayed in --list-options"
```
Comment (pmoravec) [inline tests/collect_tests/help_output_tests.py:103]: missing leading 'f' to denote f-string

File: tests/sos_tests.py [hunk 4]
```diff
@@ -600,11 +635,15 @@ def assertPluginIncluded(self, plugin):
         :type plugin:  `` str``
         """
         if not self.manifest:
-            self.error("No manifest found, cannot check for %s execution" % plugin)
+            self.error(
+                "No manifest found, cannot check for {plugin} execution"
```
Comment (pmoravec) [inline tests/sos_tests.py:639]: Same missing leading 'f'.

File: tests/sos_tests.py [hunk 5]
```diff
@@ -614,11 +653,15 @@ def assertPluginNotIncluded(self, plugin):
...
```
Comment (pmoravec) [inline tests/sos_tests.py:657]: Ditto.

**Review by pmoravec — APPROVED** [review]
ACK up to the three f-strings.

**Comment by arif-ali** [comment]
good catches, thanks, fixed those. Found a few that I didn't convert to f-strings on some of my changes, so did those too.

Thanks for all the feedback

**Review by pmoravec — APPROVED** [review]

**Pull request merged**